### ISSUE TITLE
Disk resizing issue

### ISSUE THREAD

**Issue by MikeRavenelle** [FOGProject/fogproject#418]
Reference Issue #208 

I was having issues similar to the reference issue and followed the steps to correct it. I change the d1.fixed_size_partitions from 1:2:3:4:5 to 1:2. At the end of the deployment errors were thrown in regard to chkdsk. When the image was finished being deployed it booted into windows as normal. The disk size then showed the original devices size and not the actual hard drive size of the device 30GB instead of 60GB.

The hard drive partitions were the correct size in disk management. I then ran chkdsk and extended the filesystem with diskpart. The hard drive now showed the correct size.

I would like to figure out a better solution to this issue since it is not ideal to have to run chkdsk and diskpart after every deployment.

Not sure if it is relevant but the hard drive is an NVME.

[d1.minimum.partitions.txt](https://github.com/FOGProject/fogproject/files/5810211/d1.minimum.partitions.txt)
[d1.fixed_size_partitions(current).txt](https://github.com/FOGProject/fogproject/files/5810212/d1.fixed_size_partitions.current.txt)
[d1.partitions.txt](https://github.com/FOGProject/fogproject/files/5810213/d1.partitions.txt)
[d1.fixed_size_partitions(original).txt](https://github.com/FOGProject/fogproject/files/5810214/d1.fixed_size_partitions.original.txt)

**Comment by Quazz** [comment]
What version of FOG are you using?

**Comment by Sebastian-Roth** [comment]
@MikeRavenelle There seem to be more than one issue with what you posted. First @Quazz is probably spot on when asking about the version you use because the latest version of FOG (1.5.9 as of now) is not expected to produce a d1.fixed_size_partitions of `1:2:3:4:5` - this was an issue in earlier versions though.

The other problem I see with you partition layout is that the 5th partition is kind of preventing the major sized 4th partition that you want to be expanded I suppose. As of now FOG would not move the start of a partition (sda5 in your case) further back on the disk to make room for sda4 to be expanded as much as you like it to be.

We are currently working on this and you are welcome to try out a alpha stage version of the FOG init (doing all the resize and deploy stuff). But be aware that depending on the version of FOG you currently use it's not as easy as described in the forum: https://forums.fogproject.org/topic/15025/move-partition-in-gpt-layout-need-people-to-test (let us know here if you run into issues and need help)

**Comment by MikeRavenelle** [comment]
The version of FOG that I am using is 1.5.9.

I will attempt the alpha stage version of FOG init and report back.

**Comment by Sebastian-Roth** [comment]
@MikeRavenelle said:
> The version of FOG that I am using is 1.5.9.

Oh, I did not expect that! So the alpha stage FOG init won't help you all the way through. You will still need to manually edit `d1.fixed_size_partitions` and we should really start looking into your setup more closely to find out why FOS (FOG OS doing all the work) detects all your partitions as fixed size!!

Schedule a **debug** capture task (just like when you schedule a normal one but before you hit the button in the web UI there is a checkbox for debug mode), boot up the host you want to capture from and hit ENTER twice to get to the shell. The run the following commands, take pictures of the output and post here:
```
parted -s /dev/sda print
blkid -po udev /dev/sda1 | grep FS_TYPE
blkid -po udev /dev/sda2 | grep FS_TYPE
blkid -po udev /dev/sda3 | grep FS_TYPE
blkid -po udev /dev/sda4 | grep FS_TYPE
blkid -po udev /dev/sda5 | grep FS_TYPE
```
**We need the full output** of all those commands (the first is half a dozen of lines and the later ones should be all one line of output each). Make sure you have all the output you see on screen in your picture(s).

**Comment by MikeRavenelle** [comment]
Just to clarify, do you want me to get the debug capture using the alpha FOG init? Or just a standard debug capture?

**Comment by Sebastian-Roth** [comment]
@MikeRavenelle The debug output looks fine and I am really wondering on why `sda4` is in the d1.fixed_size_partitions list?! It really shouldn't be because the FS_TYPE if correctly recognized as ntfs and the flag seen in the last column of `parted`'s output should not trigger the FOS init to mark it as fixed either.

I think we need to get that same debug output from the machine you have captured this exact image from initially.

@Quazz Can you see something here that I might overlook?

**Comment by Quazz** [comment]
@MikeRavenelle Just for clarity, was this image created on FOG 1.5.9?

From the data provided, partitions 1, 2, 3 and 5 would be marked as fixed. (partition 5 getting in the way of the disk expanding, which would be fixed with the new init)

Only thing I can imagine is that the image was captured on a version where the fixed_size marking didn't work as expected. 

You could of course try to capture a new image (or one with and one without the alpha init) and compare the results.

**Comment by MikeRavenelle** [comment]
@Sebastian-Roth I can get this debug information on the original unit that capture the image on Monday.

@Quazz I can capture images with and without the alpha init on the 64GB unit and report back.

**Comment by MikeRavenelle** [comment]
> 
> 
> @MikeRavenelle said:
> 
> > The version of FOG that I am using is 1.5.9.
> 
> Oh, I did not expect that! So the alpha stage FOG init won't help you all the way through. You will still need to manually edit `d1.fixed_size_partitions` and we should really start looking into your setup more closely to find out why FOS (FOG OS doing all the work) detects all your partitions as fixed size!!
> 
> Schedule a **debug** capture task (just like when you schedule a normal one but before you hit the button in the web UI there is a checkbox for debug mode), boot up the host you want to capture from and hit ENTER twice to get to the shell. The run the following commands, take pictures of the output and post here:
> 
> ```
> parted -s /dev/sda print
> blkid -po udev /dev/sda1 | grep FS_TYPE
> blkid -po udev /dev/sda2 | grep FS_TYPE
> blkid -po udev /dev/sda3 | grep FS_TYPE
> blkid -po udev /dev/sda4 | grep FS_TYPE
> blkid -po udev /dev/sda5 | grep FS_TYPE
> ```
> 
> **We need the full output** of all those commands (the first is half a dozen of lines and the later ones should be all one line of output each). Make sure you have all the output you see on screen in your picture(s).

Attached is an image of the output on the device where the original image was captured from.

![IMG-0285](https://user-images.githubusercontent.com/7004556/104965758-0b5c3400-59a5-11eb-9b82-86f2c7fed445.png)

**Comment by Quazz** [comment]
Based on all of that, I would say that the issue lies here: https://github.com/FOGProject/fos/blob/8d310f51d7d945c5dc1685eb6698d1aed8634dc7/Buildroot/board/FOG/FOS/rootfs_overlay/usr/share/fog/lib/funcs.sh#L466

A partition will be marked as fixed_size if the nftsresize test fails. 

Perhaps running a chkdsk on the Windows partition might help with that.

**Comment by MikeRavenelle** [comment]
After running chkdsk this was my new output from those commands.
![unnamed](https://user-images.githubusercontent.com/7004556/105097110-afa9ad80-5a6d-11eb-8537-19373e8b4903.jpg)

**Comment by Sebastian-Roth** [comment]
@Quazz Good call. Possibly the resize test fails just because there is so little space left on the device?
@MikeRavenelle How much free space do you have on /dev/sda4 (probably C: in Windows)? Which OS do you have installed?

**Comment by MikeRavenelle** [comment]
@Sebastian-Roth Sorry for the late response. It is Windows 10 LTSC and the device has around 10Gb of free space.

I went ahead and tried taking an image from the 64Gb model and it did not resize the partition either. I end up with a 59Gb image for the device. This device has around 30ish Gb of free space. I also moved partitions around on this device so that the C drive was the last partition on the drive but this also did not work for fixing the resizing issue.

An observation I made when imaging with a different devices was that it took some time for it to check if partition was resizable and then to actually resize the partition. On these particular devices that does not occur. It goes by so fast to the partclone screens I can't actually see anything.


[d1.partitions.txt](https://github.com/FOGProject/fogproject/files/5883337/d1.partitions.txt)
[d1.fixed_size_partitions.txt](https://github.com/FOGProject/fogproject/files/5883338/d1.fixed_size_partitions.txt)
[d1.minimum.partitions.txt](https://github.com/FOGProject/fogproject/files/5883339/d1.minimum.partitions.txt)



Attached are the files and screenshot for the 64Gb model I am currently working with.
![unnamed2](https://user-images.githubusercontent.com/7004556/106059771-cb8a0080-60b8-11eb-8787-6356acd3ba22.jpg)

**Comment by Sebastian-Roth** [comment]
@MikeRavenelle To me this all sounds like there is something wrong with the filesystem on those devices. May I ask you to try a **debug** capture. Go to the FOG web UI as if you would schedule a normal capture. Just before you click the last button there is a checkbox for debug. Then start the device and when it booted up you hit ENTER twice to get to a command shell. Enter the simple command `fog` and start it with ENTER. Now you can step through the whole capture with the screen waiting on every step. Please take pictures of all you see (everything before partclone) and post those here. I would hope there is something we can fine to explain what is going on.

**Comment by MikeRavenelle** [comment]
Here are the partition files for the device along with the screenshots you requested. 

[d1.minimum.partitions.txt](https://github.com/FOGProject/fogproject/files/5896074/d1.minimum.partitions.txt)
[d1.fixed_size_partitions.txt](https://github.com/FOGProject/fogproject/files/5896075/d1.fixed_size_partitions.txt)
[d1.partitions.txt](https://github.com/FOGProject/fogproject/files/5896076/d1.partitions.txt)
![IMG_0294](https://user-images.githubusercontent.com/7004556/106322450-7fb39480-623b-11eb-9c59-683dbd3bb862.jpg)
![IMG_0295](https://user-images.githubusercontent.com/7004556/106322451-7fb39480-623b-11eb-844a-768aab8769bd.jpg)
![IMG_0296](https://user-images.githubusercontent.com/7004556/106322452-7fb39480-623b-11eb-95cd-b88ad3d9b643.jpg)
![IMG_0297](https://user-images.githubusercontent.com/7004556/106322454-804c2b00-623b-11eb-9b34-953f6ce1c1b0.jpg)
![IMG_0298](https://user-images.githubusercontent.com/7004556/106322455-804c2b00-623b-11eb-9c36-df76de414740.jpg)
![IMG_0299](https://user-images.githubusercontent.com/7004556/106322458-804c2b00-623b-11eb-962f-bfffe0b65b5a.jpg)
![IMG_0300](https://user-images.githubusercontent.com/7004556/106322462-80e4c180-623b-11eb-80d9-679ea0d674fb.jpg)
![IMG_0301](https://user-images.githubusercontent.com/7004556/106322463-80e4c180-623b-11eb-8c3a-77cad52f06dd.jpg)
![IMG_0302](https://user-images.githubusercontent.com/7004556/106322464-80e4c180-623b-11eb-9296-584b01336a59.jpg)
![IMG_0303](https://user-images.githubusercontent.com/7004556/106322466-817d5800-623b-11eb-8bf0-1a5c4d9f111a.jpg)
![IMG_0304](https://user-images.githubusercontent.com/7004556/106322467-817d5800-623b-11eb-83fe-1eca89f8f84e.jpg)
![IMG_0305](https://user-images.githubusercontent.com/7004556/106322470-817d5800-623b-11eb-924c-7d2c81449c23.jpg)
![IMG_0306](https://user-images.githubusercontent.com/7004556/106322474-8215ee80-623b-11eb-9a20-cd15c24f238f.jpg)
![IMG_0307](https://user-images.githubusercontent.com/7004556/106322476-8215ee80-623b-11eb-9516-0793961e84f2.jpg)
![IMG_0308](https://user-images.githubusercontent.com/7004556/106322477-8215ee80-623b-11eb-92d7-cd510a96dfa9.jpg)
![IMG_0309](https://user-images.githubusercontent.com/7004556/106322478-82ae8500-623b-11eb-9e72-acb4f730ccd5.jpg)
![IMG_0310](https://user-images.githubusercontent.com/7004556/106322479-82ae8500-623b-11eb-9f85-3dd21f335374.jpg)
![IMG_0311](https://user-images.githubusercontent.com/7004556/106322480-82ae8500-623b-11eb-93df-28bbfdfb4b3a.jpg)
![IMG_0312](https://user-images.githubusercontent.com/7004556/106322481-82ae8500-623b-11eb-892a-d855b9e11357.jpg)
![IMG_0313](https://user-images.githubusercontent.com/7004556/106322482-83471b80-623b-11eb-916b-fbf347e86bb6.jpg)
![IMG_0314](https://user-images.githubusercontent.com/7004556/106322483-83471b80-623b-11eb-989c-7b85420b4d09.jpg)
![IMG_0315](https://user-images.githubusercontent.com/7004556/106322484-83471b80-623b-11eb-8fb9-b477847b9fa2.jpg)
![IMG_0316](https://user-images.githubusercontent.com/7004556/106322486-83471b80-623b-11eb-8e38-b236836640b2.jpg)
![IMG_0317](https://user-images.githubusercontent.com/7004556/106322487-83dfb200-623b-11eb-9ddf-7f7208ee042e.jpg)
![IMG_0318](https://user-images.githubusercontent.com/7004556/106322488-83dfb200-623b-11eb-8201-2224f07c17ab.jpg)
![IMG_0319](https://user-images.githubusercontent.com/7004556/106322490-83dfb200-623b-11eb-9be5-ce6be9e7b875.jpg)
![IMG_0320](https://user-images.githubusercontent.com/7004556/106322492-83dfb200-623b-11eb-9b7d-4cf986d72787.jpg)
![IMG_0321](https://user-images.githubusercontent.com/7004556/106322423-7c200d80-623b-11eb-9ef9-f0b4d89d8007.jpg)
![IMG_0322](https://user-images.githubusercontent.com/7004556/106322425-7cb8a400-623b-11eb-9e70-cc85fbe1546e.jpg)
![IMG_0323](https://user-images.githubusercontent.com/7004556/106322426-7cb8a400-623b-11eb-9669-1c22439a687b.jpg)
![IMG_0324](https://user-images.githubusercontent.com/7004556/106322427-7cb8a400-623b-11eb-840a-768879737219.jpg)
![IMG_0325](https://user-images.githubusercontent.com/7004556/106322428-7d513a80-623b-11eb-9b6a-ecc2b1e9b26f.jpg)
![IMG_0326](https://user-images.githubusercontent.com/7004556/106322430-7d513a80-623b-11eb-8e0b-18c37055014b.jpg)
![IMG_0327](https://user-images.githubusercontent.com/7004556/106322431-7d513a80-623b-11eb-8cc9-43528a33b204.jpg)
![IMG_0328](https://user-images.githubusercontent.com/7004556/106322433-7d513a80-623b-11eb-9053-a19ee319f5d5.jpg)
![IMG_0329](https://user-images.githubusercontent.com/7004556/106322435-7de9d100-623b-11eb-836b-30ace780098f.jpg)
![IMG_0330](https://user-images.githubusercontent.com/7004556/106322436-7de9d100-623b-11eb-8e64-97179a85a149.jpg)
![IMG_0331](https://user-images.githubusercontent.com/7004556/106322437-7de9d100-623b-11eb-9950-e4f8b9774cd2.jpg)
![IMG_0332](https://user-images.githubusercontent.com/7004556/106322438-7e826780-623b-11eb-8c70-2e922ec329a1.jpg)
![IMG_0333](https://user-images.githubusercontent.com/7004556/106322440-7e826780-623b-11eb-8637-2094b1eccd61.jpg)
![IMG_0334](https://user-images.githubusercontent.com/7004556/106322442-7e826780-623b-11eb-88a6-197cf265bca7.jpg)
![IMG_0335](https://user-images.githubusercontent.com/7004556/106322444-7e826780-623b-11eb-8ae2-37d611f87a5b.jpg)
![IMG_0336](https://user-images.githubusercontent.com/7004556/106322447-7f1afe00-623b-11eb-95d6-677654b0e2d9.jpg)
![IMG_0337](https://user-images.githubusercontent.com/7004556/106322448-7f1afe00-623b-11eb-9cfe-8991ce9719ef.jpg)
![IMG_0338](https://user-images.githubusercontent.com/7004556/106322449-7f1afe00-623b-11eb-8e4a-900946a0e37c.jpg)

**Comment by Sebastian-Roth** [comment]
@MikeRavenelle Well done!! I think @Quazz was spot on. Something goes wrong when `ntfsresize` does a first test if it's able to resize the filesystem - seen in picture https://user-images.githubusercontent.com/7004556/106322433-7d513a80-623b-11eb-9053-a19ee319f5d5.jpg

Please schedule another debug capture task and when you get to the shell run `ntfsresize -fivP /dev/sda5`, take a picture of the output and post that here.

**Comment by MikeRavenelle** [comment]
> ntfsresize -fivP /dev/sda5
![File_0001](https://user-images.githubusercontent.com/7004556/106516043-239b7b00-649c-11eb-9063-6ab0f8b31bd3.jpg)

I ran a chkdsk as instructed by the error and took an image and this was the result after part clone
![File_0000](https://user-images.githubusercontent.com/7004556/106516129-3ca42c00-649c-11eb-80ba-6134d1d1fee7.jpg)

I then tried to image this to another device and received this error.
![File_0002](https://user-images.githubusercontent.com/7004556/106516181-4c237500-649c-11eb-9b37-8efbe9e4b81c.jpg)

So I decided to run a chkdsk from windows install USB and then took an image. This seems to have resolved my issues. I was able to take the image successfully and then image it to a smaller device. 

I am not sure what the overall cause of this issue was but seems to be resolved so I will close this issue and re open if I have any other problems.
